src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -154,8 +159,19 @@ static boolean isNew(TenantAttributes attributes, String featureVersion) {
     return future.compose(result -> runJavaMigrations(attributes, vertxContext, headers).map(result));
   }
 
+  @Validate
+  public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers,
+                         Handler<AsyncResult<Response>> handler, Context context) {
+    // delete Kafka topics if tenant purged
+    Future.succeededFuture()
+      .compose(x -> tenantAttributes.getPurge()
```

Comment:
`Future.succeededFuture().compose(x -> foo)` is a noop and should be replaced by `foo`.